static/src/javascripts/projects/common/modules/atoms/youtube-player.ts

```diff
@@ -0,0 +1,303 @@
+import { onConsentChange } from '@guardian/consent-management-platform';
+import type { TCFv2ConsentState } from '@guardian/consent-management-platform/dist/types/tcfv2';
+import { loadScript } from '@guardian/libs';
+import fastdom from 'fastdom';
+import { getPageTargeting } from 'common/modules/commercial/build-page-targeting';
+import { commercialFeatures } from 'common/modules/commercial/commercial-features';
+import { buildPfpEvent } from 'common/modules/video/ga-helper';
+import config from 'lib/config';
+import { constructQuery } from 'lib/url';
+import { getPermutivePFPSegments } from '../commercial/permutive';
+
+interface WindowLocal extends Window {
+	ga: UniversalAnalytics.ga;
+	onYouTubeIframeAPIReady?: () => void;
+	YT?: typeof YT;
+}
+
+const scriptSrc = 'https://www.youtube.com/iframe_api';
+const promise = new Promise<void>((resolve) => {
+	if ((window as WindowLocal).YT?.Player) {
+		resolve();
+	} else {
+		(window as WindowLocal).onYouTubeIframeAPIReady = resolve;
+	}
+});
+
+const loadYoutubeJs = () => {
+	void loadScript(scriptSrc, {});
+};
+
+const addVideoStartedClass = (el: HTMLElement | null) => {
+	if (el) {
+		el.classList.add('youtube__video-started');
+	}
+};
+
+let tcfData: TCFv2ConsentState | undefined = undefined;
+
+interface ConsentState {
+	framework: null | string;
+	canTarget: boolean;
+}
+
+let consentState: ConsentState = {
+	framework: null,
+	canTarget: false,
+};
```

Comment:
Thank you so much for doing this. ❤️ 
Having a `consentState` object which is framework-agnostic is huge improvement.
`canTarget` is also especially intuitive and meaningful (as opposed to `ccpaState` and `tcfState`).